internal/pkg/db/repository/repository_room.go

```diff
@@ -0,0 +1,340 @@
+package repository
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"time"
+
+	"github.com/volatiletech/null/v8"
+	"github.com/volatiletech/sqlboiler/v4/boil"
+	"github.com/volatiletech/sqlboiler/v4/queries/qm"
+
+	"github.com/FurmanovD/roommgr/internal/pkg/db/automodel"
+	"github.com/FurmanovD/roommgr/pkg/sqldb"
+)
+
+type roomRepositoryImpl struct {
+	db sqldb.SqlDB
+}
+
+// ================= interface instantiation =================================================
+
+func NewRoomRepository(db sqldb.SqlDB) RoomRepository {
+	return &roomRepositoryImpl{
+		db: db,
+	}
+}
+
+// ================= interface methods =================================================
+func (r *roomRepositoryImpl) GetRoom(ctx context.Context, id int) (*automodel.Room, error) {
+
+	room, err := automodel.Rooms(
+		automodel.RoomWhere.ID.EQ(id),
+	).One(ctx, r.db.Connection())
+
+	if err != nil {
+		if err == sql.ErrNoRows {
+			return nil, nil
+		}
+		return nil, err
+	}
+
+	return room, nil
+}
+
+func (r *roomRepositoryImpl) GetRooms(ctx context.Context, filter RoomFilter) (automodel.RoomSlice, error) {
+
+	var qm []qm.QueryMod
+	if !filter.IncludeDeleted {
+		qm = append(qm, automodel.RoomWhere.DeletedAt.EQ(null.TimeFromPtr(nil)))
+	}
+	rooms, err := automodel.Rooms(
+		qm...,
+	).All(ctx, r.db.Connection())
+
+	if err != nil {
+		if err == sql.ErrNoRows {
```

Comment:
comparing with == will fail on wrapped errors. Use errors.Is to check for a specific error